File: sktg/features/misc.py
from ..telegram import *


@command("source", "opensource", "github")
async def github_link(message: types.Message):
    return await message.reply(
        "https://github.com/SandaruKasa/SKTG/tree/python",
        disable_web_page_preview=True,
    )


@command("shrug")
async def shrug(message: types.Message):
    return await message.reply(r"¯\_(ツ)_/¯")


@command("uptime", filters=(bot_admin_filter,))
async def uptime(message: types.Message):
    result = get_uptime()
    if result is None:
        result = "Unknown"
    else:
        # stripping microseconds
        result = str(result).split(".")[0]
    return await message.reply(text=result)

File: src/features/misc.rs
use teloxide::{prelude::*, types::Message, utils::command::BotCommands};

use crate::types::*;

#[derive(BotCommands, Clone)]
#[command(rename = "lowercase", description = "")]
pub enum MiscCommands {
    #[command(description = "off")]
    Shrug,
    #[command(description = "off")]
    GitHub,
    #[command(description = "off")]
    Source,
}

use MiscCommands::*;

pub async fn misc(bot: TelegramBot, message: Message, command: MiscCommands) -> Res<()> {
    bot.send_message(
        message.chat.id,
        match command {
            Shrug => "¯\\_(ツ)_/¯",
            GitHub | Source => "https://github.com/SandaruKasa/SKTG/tree/rust",
        },
    )
    .reply_to_message_id(message.id)
    .disable_web_page_preview(true)
    .await?;
    Ok(())
}
